environments/dotnet/ExecutorModule.cs

```diff
@@ -0,0 +1,65 @@
+using Fission.DotNetCore.Compiler;
+using System.Collections.Generic;
+using Nancy;
+using System.IO;
+using System;
+
+namespace Fission.DotNetCore
+{
+    public class ExecutorModule : NancyModule
+    {
+        private const string CODE_PATH = "/userfunc/user";
+
+        private static Function _userFunc;
+
+        public ExecutorModule()
+        {
+            Post("/specialize", args => Specialize());
+            Get("/", _ => Run());
+            Post("/", _ => Run());
+            Put("/", _ => Run());
+            Head("/", _ => Run());
+            Options("/", _ => Run());
+            Delete("/", _ => Run());
+
+        }
+
+        private object Specialize()
+        {
+            var errors = new List<string>();
+            if (File.Exists(CODE_PATH))
+            {
+                var code = File.ReadAllText(CODE_PATH);
+                _userFunc = FissionCompiler.Compile(code, out errors);
+                if (_userFunc == null)
+                {
+                    var errstr = string.Join(Environment.NewLine, errors);
+                    Console.WriteLine(errstr);
+                    var response = (Response)errstr;
+                    response.StatusCode = HttpStatusCode.InternalServerError;
+                    return response;
+
+                }
+                return null;
+            }
+            else
+            {
+                var response = (Response)"Unable to locate code";
+                response.StatusCode = HttpStatusCode.InternalServerError;
+                return response;
+            }
+        }
+
+        private object Run()
+        {
+            if (_userFunc == null)
+            {
+                var response = (Response)"Generic container: no requests supported";
+                response.StatusCode = HttpStatusCode.InternalServerError;
+                return response;
+            }
+            var args = ((DynamicDictionary)Request.Query).ToDictionary();
```

Comment:
Thanks for adding the context class.  

With the other environments (nodejs, python) we did not abstract away HTTP.  So functions can access the full request object in both the python and nodejs environments.  This was a conscious decision not just to reduce the complexity of the environments, but also to make it easier for function programmers to gather whatever info they want from the request.  The web frameworks being used in those cases have stable and well documented APIs.

But, both those are dynamic/duck-typed languages so some of the concerns you're raising about declarations haven't come up before.  I think one way to go would be to have the query string args easily parsed out as a dictionary, and the rest of the request object available to the function through some method.  That way the most common use cases simply use the context.args dictionary and don't need any using declarations for nancy.  What do you think?

For logging, stdout is good.  The plan is to aggregate all logs from stdout into one of log aggregation systems.  (stdout is also what shows up in `kubectl logs`, so it's good for users too.)